tests/helpers/test_mongo.py

```diff
@@ -81,6 +82,42 @@ def test_get_positive_samples_in_source_plate_returns_none_failure_fetching_samp
             assert result is None
 
 
+def test_get_all_samples_for_source_plate_returns_matching_samples(app, samples):
+    with app.app_context():
+        samples, _ = samples
+        source_plate_uuid = samples[0][FIELD_LH_SOURCE_PLATE_UUID]
+        expected_samples = list(
+            filter(lambda x: x.get(FIELD_LH_SOURCE_PLATE_UUID, False) == source_plate_uuid, samples)
+        )
+        result = get_all_samples_for_source_plate(source_plate_uuid)
+
+        # remove the microsecond before comparing
+        if result is not None:
+            for res in result:
+                res.update({FIELD_DATE_TESTED: res[FIELD_DATE_TESTED].replace(microsecond=0)})
+            for sample in expected_samples:
+                sample.update({FIELD_DATE_TESTED: sample[FIELD_DATE_TESTED].replace(microsecond=0)})
+
+        assert result == expected_samples
+
+
+def test_get_all_samples_for_source_plate_returns_empty_list_no_matching_samples(app, samples):
+    with app.app_context():
+        result = get_all_samples_for_source_plate("source plate uuid does not exist")
+
+        assert result == []
```

Comment:
We can add a test when the argument is None

Reply:
I've done this and pushed up the change.  I think we probably should revert that last commit though.  Look at it and tell me what you think.  mypy is already protecting from this issue occurring through the typing system.